### PR TITLE
Finalize split preparation-2

### PR DESCRIPTION
**Pull request by ronald-cron-arm**
## Description
Progresses #9624 

## PR checklist
- [x] **changelog** not required because: no change from the user perspective 
- [x] **development PR** here
- [x] **framework PR** provided Mbed-TLS/mbedtls-framework#99
- [x] **3.6 PR** provided #9832, necessary because of a change in the framework
- [x] **2.28 PR** provided # | not required because: 4.0 only work
- **tests**  not required because: just changing the way things are done

### CODE REVIEW

File: tests/scripts/components-configuration-crypto.sh [hunk 1]
```diff
@@ -2314,6 +2309,11 @@ END
         #define PSA_WANT_ALG_SHA3_256  1
```
Comment (minosgalanakis) [inline tests/scripts/components-configuration-crypto.sh:2300]: Any reason we do not create a configuration file for this?

File: tests/configs/tls13-only.h [hunk 2]
```diff
@@ -26,6 +26,3 @@
 #undef MBEDTLS_SSL_DTLS_CLIENT_PORT_REUSE
 #undef MBEDTLS_SSL_DTLS_CONNECTION_ID
 #undef MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT
-
-/* Enable some invasive tests */
-#define MBEDTLS_TEST_HOOKS
```
Comment (minosgalanakis) [inline tests/configs/tls13-only.h:31]: Not a blocker, but I am curious as to why moving is to a component set parameter was required. All tls12_only tests need to asser it.

File: tests/scripts/components-configuration-crypto.sh [hunk 3]
```diff
@@ -2167,6 +2158,8 @@ component_test_ccm_aes_sha256 () {
 
     # Setting a blank config disables everyhing in the library side.
     echo '#define MBEDTLS_CONFIG_H ' >"$CONFIG_H"
+    cp configs/crypto-config-ccm-aes-sha256.h "$CRYPTO_CONFIG_H"
```
Comment (minosgalanakis) [inline tests/scripts/components-configuration-crypto.sh:2161]: Nice catch, thanks.

**Review by minosgalanakis — DISMISSED** [review]
Looks good overall, minor tibits/comments.

**Review by gabor-mezei-arm — DISMISSED** [review]
The CI is not happy with the generated files. Otherwise looks good to me.

File: tests/scripts/components-configuration-crypto.sh (hunk 3 above)
Comment (ronald-cron-arm) [reply tests/scripts/components-configuration-crypto.sh:2161]: Thanks. I've compared the outcomes.csv from before and after #9567 and it was quite clear there was an issue with test_ccm_aes_sha256.

File: tests/configs/tls13-only.h (hunk 2 above)
Comment (ronald-cron-arm) [reply tests/configs/tls13-only.h:31]: tls13-only.sh is an mbedtls user config file and MBEDTLS_TEST_HOOKS now belongs to crypto_config.h. That's the purpose of the change, having MBEDTLS_TEST_HOOKS defined in crypto_config.h not mbedtls_config.h.

**Comment by ronald-cron-arm** [comment]
@minosgalanakis @gabor-mezei-arm CI is green now, please have another look, thanks.

**Review by gabor-mezei-arm — APPROVED** [review]
LGTM

**Review by minosgalanakis — DISMISSED** [review]
LGTM

**Review by gabor-mezei-arm — APPROVED** [review]
LGTM

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Pull request merged**